bundles-lib/src/main/java/org/apache/metron/bundles/ExtensionMapping.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.bundles;
+
+import org.apache.metron.bundles.bundle.BundleCoordinates;
+
+import java.util.*;
+import java.util.function.BiFunction;
+
+public class ExtensionMapping {
+
+  private final Map<String, Map<String, Set<BundleCoordinates>>> extensionNameMap = new HashMap<>();
```

Comment:
So - the goal with the Extension* stuff was to remove the necessity to statically declare the classes of extensions used.  They are passed in to the ExtensionManager and the BundleThreadContextClassLoader.   The ExtensionMapping is different, it ends up getting the classes from discover in the nifi properties, where the classes are defined, and typed.
So yes, the type does subdivide the namespace.
Type -> Classname -> coordinates
Parser -> MessageParser -> Parser Extension Coordinates